pkg/apiclient/client_factory.go

```diff
@@ -40,6 +41,9 @@ type ClientFactory interface {
 
 	// GetHostUrl returns the current set API URL as a string
 	GetHostUrl() string
+
+	// GetHttpClient returns a raw http client which can be used to query Octopus
+	GetHttpClient() (*http.Client, error)
```

Comment:
We don't use `SystemClient` from the go-octopusdeploy library here because we need to send anonymous requests to the /.well-known/openid-configuration endpoint on the Octopus Server and then follow to the token endpoint. The system client assumes a base path of /api which doesn't apply here. 

So this function has been added to the factory to allow the login command to send the relevant requests but also allow us to mock these in tests.